Common/util/string_utils.cpp

```diff
@@ -215,6 +217,15 @@ void StrUtil::ReadString(char *cstr, Stream *in, size_t buf_limit)
         cstr[0] = 0;
 }
 
+void StrUtil::ReadString(String *s, Stream *in)
+{
+    size_t len = in->ReadInt32();
+    s->FillString('\0',len);
+    char* ptr = (char*)s->GetCStr(); //yes, this is bad.
+    if (len > 0)
+        in->Read(ptr, len);
+}
+
```

Comment:
I think you do not need this function. There's another variant of ReadString above that returns String object:
<pre>
String StrUtil::ReadString(Stream *in)
{
    size_t len = in->ReadInt32();
    if (len > 0)
        return String::FromStreamCount(in, len);
    return String();
}
</pre>

It works with String::ReadCount which ensures null-terminator (well, supposed to).

Reply:
OK ill change this to use that, but I should leave the function because there are functions there for reading every other string type in that way and the whole reason I added the function is because it seemed like it should be there, but wasn't. Someone else will add the function on accident again one day otherwise